### ISSUE TITLE
pushAndRemoveUntil not play push animation if remove all except the first route

### ISSUE THREAD

**Issue by ivk1800** [flutter/flutter#25080]
## Steps to Reproduce

1. run example
2. tap "push" button 
3. tap "pushAndRemoveUntil" button
4. push animation not played

example:
```dart
import 'package:flutter/material.dart';
import 'dart:math';

void main() {
  runApp(MaterialApp(
    home: Page(),
  ));
}

class Page extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Color((Random().nextDouble() * 0xFFFFFF).toInt() << 0)
          .withOpacity(1.0),
      appBar: AppBar(
        title: Text(new DateTime.now().millisecondsSinceEpoch.toString()),
      ),
      body: Column(
        children: <Widget>[
          RaisedButton(
            child: Text('pushAndRemoveUntil'),
            onPressed: () {
              var navigator = Navigator.of(context);

              var route = MaterialPageRoute(
                  builder: ((BuildContext context) => Page()));
              navigator.pushAndRemoveUntil(route, (Route<dynamic> route) {
                print(route);
                return route.isFirst;
              });
            },
          ),
          RaisedButton(
            child: Text('push'),
            onPressed: () {
              var navigator = Navigator.of(context);

              navigator.push(
                  MaterialPageRoute(
                      builder: (BuildContext context) => Page()));
            },
          ),
        ],
      ),
    );
  }
}
```

![not_animation_route](https://user-images.githubusercontent.com/11444783/49651475-b9682680-fa48-11e8-883c-5553cea041ac.gif)

doctor

```
Flutter (Channel stable, v1.0.0, on Mac OS X 10.14 18A391, locale ru-RU)
    • Flutter version 1.0.0 at /Users/ik/Repositories/flutter
    • Framework revision 5391447fae (7 days ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[!] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/ik/Library/Android/sdk
    • Android NDK at /Users/ik/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1024-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[!] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    ✗ Verify that all connected devices have been paired with this computer in
      Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may
      require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2018.2.6)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 30.0.2
    • Dart plugin version 182.5124

[✓] Connected device (2 available)
    • iPhone 6  • D0700405-7A5D-498F-A980-BC6D4F1E9E46 • ios • iOS 12.0
      (simulator)
    • iPhone SE • 91A75C13-5F1C-4551-9917-5848E90CF772 • ios • iOS 12.0
      (simulator)
```

**Comment by jeffwoooo** [comment]
I have same issue too. Just figured out with pushing dummy widget as first route, but so awkward to put meaningless widget on my app root.

**Comment by jeffwoooo** [comment]
I have same issue. Wish there's an alternative way(like manually popping behind the scene after pushing) but seems there isn't.

**Comment by pepegich** [comment]
@jeffwoooo i fixed it https://github.com/flutter/flutter/pull/28727

**Comment by Asoseil** [comment]
some news about this?

**Comment by goderbauer** [comment]
https://github.com/flutter/flutter/pull/30211 should fix this.

**Comment by Xgamefactory** [comment]
still no solution.

**Comment by Asoseil** [comment]
It still doesn't work on stable channel v1.7

**Comment by Piinks** [comment]
This patch has not yet landed on `stable`. It is currently on the `master` branch, and should be making its way to the `dev` branch next. The [Flutter build release channels](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels) wiki page has more information if you would like to learn more about Flutter's channels and release cadence. :)

**Comment by Nico04** [comment]
How do you know THIS issue is not on the stable release?
How do you know on which release it is? 
Thanks

**Comment by Piinks** [comment]
Good question @Nico04!
If you refer to the pull request that  closed the issue, in this case #35223, and click on the commit that that was made to master when it was merged, 97b2c98642c1ec7e3427f616a82b45b287f5f2b2, you will see a list of tags at the end of the commit description. 
These tags specify the channels and versions that this commit is included in.

Here is the fix in question with the master tag circled below:
![Screen Shot 2019-07-12 at 4 05 10 PM](https://user-images.githubusercontent.com/16964204/61162783-2e0fb180-a4bf-11e9-8b42-3920574c71b1.png)

And for demonstration, here is a much older commit that has tags specifying the versions it is included in:
![Screen Shot 2019-07-12 at 4 10 37 PM](https://user-images.githubusercontent.com/16964204/61162992-47652d80-a4c0-11e9-8aca-aa77f3e20bda.png)
